client-server-client/src/main/scala/zio/raft/client/RaftClient.scala

```diff
@@ -0,0 +1,461 @@
+          for {
+            requestId <- nextRequestId.getAndUpdate(_.next)
+            now <- Clock.instant
+            request = ClientRequest(requestId, payload, now)
+            _ <- transport.sendMessage(request).orDie
+          } yield copy(pendingRequests = pendingRequests.updated(requestId, promise))
+        
+        case StreamEvent.ServerMsg(ClientResponse(requestId, result)) =>
+          pendingRequests.get(requestId) match {
+            case Some(promise) =>
+              promise.succeed(result).ignore.as(
+                copy(pendingRequests = pendingRequests.removed(requestId))
+              )
+            case None =>
+              ZIO.logWarning(s"Response for unknown request: $requestId").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(KeepAliveResponse(_)) =>
+          ZIO.succeed(this)
+        
+        case StreamEvent.ServerMsg(serverRequest: ServerRequest) =>
+          for {
+            // Enqueue for user to handle
+            _ <- serverRequestQueue.offer(serverRequest).orDie
+            // Immediately acknowledge
+            _ <- transport.sendMessage(ServerRequestAck(serverRequest.requestId)).orDie
+            _ <- ZIO.logDebug(s"Enqueued and acknowledged server request: ${serverRequest.requestId}")
+          } yield this
+        
+        case StreamEvent.ServerMsg(RequestError(reason, leaderId)) =>
+          reason match {
+            case NotLeaderRequest =>
+              for {
+                _ <- ZIO.logInfo(s"Not leader, reconnecting")
+                nonce <- Nonce.generate()
+                _ <- transport.sendMessage(ContinueSession(sessionId, nonce)).orDie
+              } yield ConnectingExistingSession(
+                sessionId = sessionId,
+                capabilities = capabilities,
+                nonce = nonce,
+                addresses = addresses,
+                currentAddressIndex = 0,
+                nextRequestId = nextRequestId,
+                pendingRequests = pendingRequests
+              )
+            
+            case SessionTerminated =>
+              ZIO.logWarning("Session terminated").as(Disconnected)
+            
+            case _ =>
+              ZIO.logWarning(s"Request error: $reason").as(this)
+          }
+        
+        case StreamEvent.ServerMsg(SessionClosed(reason, _)) =>
+          ZIO.logInfo(s"Session closed: $reason").as(Disconnected)
+        
+        case StreamEvent.KeepAliveTick =>
+          for {
+            now <- Clock.instant
+            _ <- transport.sendMessage(KeepAlive(now)).orDie
+          } yield this
+        
+        case _ =>
+          ZIO.succeed(this)
+      }
+    }
+  }
+}
+
+private sealed trait StreamEvent
+
+private object StreamEvent {
+  case class Action(action: ClientAction) extends StreamEvent
+  case class ServerMsg(message: ServerMessage) extends StreamEvent
+  case object KeepAliveTick extends StreamEvent
+}
+
+trait ZmqClientTransport {
```

Comment:
we also need a disconnect and connect methods on the transport, accepting the address.